pandas/core/indexes/category.py

```diff
@@ -697,3 +697,15 @@ def _delegate_method(self, name: str, *args, **kwargs):
         if is_scalar(res):
             return res
         return CategoricalIndex(res, name=self.name)
+
+    def replace(
+        self,
+        to_replace=None,
+        value=None,
+        limit=None,
+        regex=False,
+        method="pad",
+    ):
+        raise NotImplementedError(
+            "Replacing values of a CategoricalIndex is not supported."
+        )
```

Comment:
remind me why we need to disable this?

Reply:
Otherwise it hits #37899.

Reply:
What do you think?

Reply:
So some cases will hit the same bug that affects Series.replace and other cases will work fine (assuming we don't raise here)?  If so, then we're better off matching Series behavior and allowing the subset of cases that do work.

Reply:
I disabled `regex=` due to #38447 . Rest is enabled and added tests are added.